Mathlib/Data/Nat/GCD/Basic.lean

```diff
@@ -345,4 +345,49 @@ theorem Coprime.mul_add_mul_ne_mul {m n a b : ℕ} (cop : Coprime m n) (ha : a 
   rw [← mul_assoc, ← h, add_mul, add_mul, mul_comm _ n, ← mul_assoc, mul_comm y]
 #align nat.coprime.mul_add_mul_ne_mul Nat.Coprime.mul_add_mul_ne_mul
 
+variable {x n m : ℕ}
+
+theorem dvd_gcd_mul_of_dvd_mul  (h : x ∣ n * m) : x ∣ gcd x n * m := by
+  obtain ⟨y, hy⟩ := h
+  rw [← gcd_mul_right, hy, gcd_mul_left]
+  exact dvd_mul_right x (gcd m y)
+
+theorem dvd_mul_of_dvd_gcd_mul (h : x ∣ gcd x n * m) : x ∣ n * m := by
+  obtain ⟨y, hy⟩ := h
+  use y * (n / gcd x n)
+  rw [← mul_assoc, ← hy, mul_comm (_ * m), ← mul_assoc, Nat.div_mul_cancel (gcd_dvd_right x n)]
+
+theorem dvd_gcd_mul_iff_dvd_gcd : x ∣ gcd x n * m ↔ x ∣ n * m :=
+  ⟨dvd_mul_of_dvd_gcd_mul, dvd_gcd_mul_of_dvd_mul⟩
+
+theorem dvd_mul_gcd_of_dvd_mul (h : x ∣ n * m) : x ∣ n * gcd x m := by
+  obtain ⟨y, hy⟩ := h
+  rw [← gcd_mul_left, hy, mul_comm, gcd_mul_left]
+  exact dvd_mul_right x (gcd n y)
+
+theorem dvd_mul_of_dvd_mul_gcd (h : x ∣ n * gcd x m) : x ∣ n * m := by
+  obtain ⟨y, hy⟩ := h
+  use y * (m / gcd x m)
+  rw [← mul_assoc, ← hy, mul_assoc, mul_comm (gcd x m), Nat.div_mul_cancel (gcd_dvd_right x m)]
+
+theorem dvd_mul_gcd_iff_dvd_gcd : x ∣ n * gcd x m ↔ x ∣ n * m :=
```

Comment:
remove the previous two lemmas, and prove this using `dvd_gcd_mul_iff_dvd_gcd` and `mul_comm`.